sw/device/sca/aes_serial.c

```diff
@@ -136,10 +120,16 @@ static void aes_key_config(const uint8_t *key, size_t key_len) {
 static void aes_key_mask_and_config(const uint8_t *key, size_t key_len) {
   SS_CHECK(key_len == kAesKeyLength);
   dif_aes_key_share_t key_shares;
-  prng_rand_bytes((uint8_t *)key_shares.share1, key_len);
+  // Mask the provided key.
   for (int i = 0; i < key_len / 4; ++i) {
+    key_shares.share1[i] = next_lfsr();
     key_shares.share0[i] = *((uint32_t *)key + i) ^ key_shares.share1[i];
   }
+  // Provide random shares for unused key bits.
+  for (int i = key_len; i < kAesKeyLengthMax / 4; ++i) {
+    key_shares.share1[i] = next_lfsr();
+    key_shares.share0[i] = next_lfsr();
```

Comment:
Shouldn't this be `key_shares.share0[i] = key_shares.share1[i];` if we want to initialise the unused bits to 0.
Or is it OK to initialise them to a random value?